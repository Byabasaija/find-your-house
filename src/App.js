import { Route, withRouter } from 'react-router-dom';
import React from 'react';
import Login from './auth/Login';
import Signup from './auth/Signup';
import Houses from './containers/Houses';
import HouseDetails from './containers/HouseDetails';
import UserFavorites from './containers/UserFavorites';
// import Header from './components/Header';

function App() {
  // const authenticatedUser = () => (
  //   <Switch>
  //     <Route exact path="/" component={Houses} />
  //     <Route exact path="/house/:id" component={HouseDetails} />
  //     <Route exact path="/my-favorites" component={UserFavorites} />
  //     <Route path="/signup" component={Signup} />
  //   </Switch>
  // );

  // const notAuthenticated = () => (
  //   <Switch>
  //     <Route exact path="/" component={Login} />
  //     <Route path="/signup" component={Signup} />
  //   </Switch>
  // );

  // const renderRoutes = () => {
  //   if (token) {
  //     return authenticatedUser();
  //   }
  //   return notAuthenticated();
  // };

  return (
    <div>
      <Route exact path="/" component={Houses} />
      <Route exact path="/house/:id" component={HouseDetails} />
      <Route exact path="/my-favorites" component={UserFavorites} />
      <Route path="/login" component={Login} />
      <Route path="/signup" component={Signup} />
    </div>
  );
}

export default withRouter(App);
