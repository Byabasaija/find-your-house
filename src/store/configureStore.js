import { applyMiddleware, createStore, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/index';

const saveToLocalStorage = (state) => {
  try {
    const serialisedState = JSON.stringify(state);
    localStorage.setItem('persistantState', serialisedState);
  } catch (e) {
    console.warn(e);
  }
};
const loadFromLocalStorage = () => {
  try {
    const serialisedState = localStorage.getItem('persistantState');
    if (serialisedState === null) return undefined;
    return JSON.parse(serialisedState);
  } catch (e) {
    console.warn(e);
    return undefined;
  }
};

const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(rootReducer, loadFromLocalStorage(),
  composeEnhancer(applyMiddleware(thunk)));

store.subscribe(() => saveToLocalStorage(store.getState()));

export default store;
