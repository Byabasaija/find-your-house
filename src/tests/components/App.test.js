import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store/configureStore';
import App from '../../App';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
