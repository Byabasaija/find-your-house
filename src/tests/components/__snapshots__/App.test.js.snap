// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="limiter"
  >
    <div
      className="container-login100 login-pg"
    >
      <div
        className="wrap-login100 p-t-190 p-b-30"
      >
        <div
          className="p-5"
        >
          <h2
            className="login-text login100-form-title p-t-20 p-b-45 pb-3"
          >
            Login
          </h2>
          <p
            className="login-text login100-form-text"
          >
            Hey there! Welcome to back
          </p>
        </div>
        <div>
          <form
            className="login100-form validate-form"
            onSubmit={[Function]}
          >
            <div
              className="form-group pb-3 wrap-input100 validate-input m-b-10"
            >
              <input
                className="form-control p-2 input100"
                name="username"
                onChange={[Function]}
                placeholder="Name..."
                required={true}
                value=""
              />
            </div>
            <div
              className="form-group pb-3 wrap-input100 validate-input m-b-10"
            >
              <input
                className="form-control p-2 input100"
                name="password"
                onChange={[Function]}
                placeholder="Password..."
                required={true}
                type="password"
                value=""
              />
            </div>
            <div
              className="container-login100-form-btn p-t-10"
            >
              <button
                className="login100-form-btn"
                onSubmit={[Function]}
                type="submit"
              >
                Log in
              </button>
            </div>
          </form>
          <div
            className="text-center w-full"
          >
            <a
              className="text-dark"
              href="/signup"
              onClick={[Function]}
            >
              Create new account
              <i
                className="fa fa-long-arrow-right"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
